code/modules/paperwork/paperbin.dm

```diff
@@ -15,19 +15,33 @@
 	var/total_paper = 30
 	var/list/papers = list()
 	var/obj/item/pen/bin_pen
+	///Overlay of the pen on top of the bin.
+	var/mutable_appearance/pen_overlay
+	///Name of icon that goes over the paper overlays.
+	var/bin_overlay_string = "paper_bin_overlay"
+	///Overlay that goes over the paper overlays.
+	var/mutable_appearance/bin_overlay
 
 /obj/item/paper_bin/Initialize(mapload)
 	. = ..()
 	interaction_flags_item &= ~INTERACT_ITEM_ATTACK_HAND_PICKUP
 	AddElement(/datum/element/drag_pickup)
-	if(!mapload)
-		return
-	var/obj/item/pen/P = locate(/obj/item/pen) in src.loc
-	if(P && !bin_pen)
-		P.forceMove(src)
-		bin_pen = P
-		update_appearance()
-	
+	if(mapload)
+		var/obj/item/pen/P = locate(/obj/item/pen) in src.loc
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Also, don't use `src.`. I know this is old code, but you should touch it up while you're here.